test/functional/page_objects/settings_page.ts

```diff
@@ -787,7 +787,12 @@ export class SettingsPageObject extends FtrService {
   ) {
     await this.clickAddField();
     await this.setFieldName(name);
-    await this.setFieldTypeComposite();
+
+    await this.retry.try(async () => {
+      await this.setFieldType('Composite');
+      const comboBox = await this.testSubjects.find('typeField');
+      expect(await this.comboBox.isOptionSelected(comboBox, 'Composite')).to.be(true);
+    });
```

Comment:
What do you think about moving this check and retry logic into the `this.setFieldType` method? That way the stability fix would apply to all field type selections and not just to `Composite`.